### ISSUE TITLE
[BUG] Previous header name should be seen when user clicks on < from in app browser after navigated to different link

### ISSUE THREAD

**Issue by ValoraQA** [valora-inc/wallet#2339]
**Frequency :** 100%

**Repro on build version:** iOS Test Flight Release Build V:1.30.1
**Repro on device :** iPhone 12(14.7.1), iPhone 13 mini(13.1.1)

**Does not Repro on build version:** Android Internal Release Build V:1.30.1
**Does not Repro on device :** Google Pixel 2XL(11.0), Redmi Note 7 Pro(10.0)

**Pre-condition:**
1] User must have downloaded the application 
2] user must have completed onboarding flow
3] User must be on homepage
4] User must be on Dapps page

**Repro Steps:**
1] Tap on a dapp (Eg : ubeswap) and open in in-app browser
2] Tap on hamburger menu and select any link
3] Observe the header name along with the URL
4] Now tap on < from bottom navigation 
5] User will be redirected back to previous page
6] Observe the header name 

**Bug :** When user navigates back to previous page, header name is shown of recently visited page
**Eg :** If user navigates to bridge link, then clicks on < from footer , navigates back to homepage of ubeswap but header name is seen as bridge
 
**Expected Behaviour :** Ideally the header name should change as per the page on which we are navigated

**Impact :** User might get confused as header name is different and the page seen is different 

**Attachment :** [header.mp4](https://drive.google.com/file/d/1o5451Y00rggJoeeakix_arnkcO6jENuC/view?usp=sharing)

**Comment by MuckT** [comment]
@kathaypacific would you be able to take a look at this?

**Comment by kathaypacific** [comment]
hey @MuckT to update you, i took a look today but seems like a bug with the react-native-webview lib (the previous title is being returned by the `onNavigationStateChange` callback, seems to be happening only on iOS devices - i could not even repro on the simulator) - i tried a few things but none worked, will keep looking tomorrow.

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified the above issue on Latest Test Flight Release Build V1.32.0 & observed the following:

**Observation:**
- "Loading" title is seen continuously when user redirects to previous page while tapping on < from in app browser after navigated to different link

**Devices:** iPhone 13 mini(15.1.1), iPhone 12(14.6.1)

**Attachment:** [Loading dapp.mp4](https://drive.google.com/file/d/1Q2bMA2MgFfFqwrJx8rshtNM9eBVEG6iN/view?usp=sharing)

Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified the above issue on Latest Test Flight Release Build V1.32.0 & observed the following:

**Observation:**
- "Loading" title is seen continuously when user redirects to previous page while tapping on < from in app browser after navigated to different link

**Devices:** iPhone 13 mini(15.1.1), iPhone 12(14.6.1)

**Attachment:** [Loading dapp.mp4](https://drive.google.com/file/d/1Q2bMA2MgFfFqwrJx8rshtNM9eBVEG6iN/view?usp=sharing)

Thanks..!!

**Comment by kathaypacific** [comment]
Hi @ValoraQA , the behaviour observed above is expected. there is a technical note that I left in the PR description that you maybe missed :) it says `We are using onNavigationStateChange to update the screen header information. This seems to be [unreliable](https://stackoverflow.com/a/50137971) depending on the implementation of the particular web page displayed. A symptom of this problem is outlined in https://github.com/valora-inc/wallet/issues/2339 but it does not affect all dapps (navigation back from external links in dapps like Revo/Moola/Good Ghosting did not repro this behaviour). This PR ensures that at least there will not be a mismatch in page title/url displayed, on Ubeswap the "Loading..." title may be displayed on navigating back from an external link but will be reloaded on the next user actions.`

in summary, it is not possible to fix this bug completely but what we can do is prevent the header title and url from becoming out of sync. In the observation above, the "loading..." title will be corrected when you navigate to another place in Ubeswap (e.g. by clicking on another header item)

please confirm if there are any further concerns, or close this issue if not :) thanks!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified the above issue on Latest Test Flight Release Build V1.32.0 & observed that the "Loading" title is seen continuously when user redirects to previous page and when user navigates to another link then loading text changes to the link text , hence we are closing the issue

**Devices:** iPhone 13 mini(15.1.1), iPhone 12(14.6.1)

**Attachment:** [Loading dapp.mp4](https://drive.google.com/file/d/1ApbLbG3edfjYtmaNhW_wl9UXSGWGS-9E/view?usp=sharing)

Thanks..!!
